docs/vscode_sonarlint_integration.md

```diff
@@ -0,0 +1,28 @@
+# SonarTS and SonarLint Visual Studio Code integration
+
+[SonarLint for Visual Code](https://www.sonarlint.org/vscode/index.html) integrates SonarTS to provide on-the-fly 
+analysis for TypeScript files. This document aims to explain mechanics of this integration.
+
+## Startup
+
+When VSCode activates SonarLint extension and requests analysis for the TypeScript file, first SonarLint Language Server 
+is started. Communication between VSCode extension and SonarLint Language Server is using 
```

Comment:
@saberduck you call the same in 2 different ways: `VSCode extension` and `SonarLint extension` on previous line. Let's use the same name as in the diagram